test/sample-test/sample_test_launcher.py

```diff
@@ -0,0 +1,262 @@
+PAPERMILL_ERR_MSG = 'An Exception was encountered at'
+
+
+#TODO(numerology): Add unit-test for classes.
+class SampleTest(object):
+  """Launch a KFP sample_test provided its name.
+
+  Args:
+    test_name: name of the sample test.
+    input: The path of a pipeline package that will be submitted.
+    result: The path of the test result that will be exported.
+    output: The path of the test output.
+    namespace: Namespace of the deployed pipeline system. Default: kubeflow
+  """
+
+  GITHUB_REPO = 'kubeflow/pipelines'
+  BASE_DIR= '/python/src/github.com/' + GITHUB_REPO
+  TEST_DIR = BASE_DIR + '/test/sample-test'
+
+  def __init__(self, test_name, results_gcs_dir, target_image_prefix='',
+                         namespace='kubeflow'):
+    self._test_name = test_name
+    self._results_gcs_dir = results_gcs_dir
+    self._target_image_prefix = target_image_prefix
+    self._namespace = namespace
+    self._sample_test_result = 'junit_Sample%sOutput.xml' % self._test_name
+    self._sample_test_output = self._results_gcs_dir
+    self._work_dir = self.BASE_DIR + '/samples/core/' + self._test_name
+
+  def check_result(self):
+    os.chdir(self.TEST_DIR)
+    subprocess.call([
+        sys.executable,
+        'run_sample_test.py',
+        '--input',
+        '%s/%s.yaml' % (self._work_dir, self._test_name),
+        '--result',
+        self._sample_test_result,
+        '--output',
+        self._sample_test_output,
+        '--testname',
+        self._test_name,
+        '--namespace',
+        self._namespace
+    ])
+    print('Copy the test results to GCS %s/' % self._results_gcs_dir)
+    working_bucket = PROJECT_NAME
+
+    utils.upload_blob(
+        working_bucket,
+        self._sample_test_result,
+        self._results_gcs_dir + '/' + self._sample_test_result
+    )
+
+  def check_notebook_result(self):
+    # Workaround because papermill does not directly return exit code.
+    exit_code = '1' if PAPERMILL_ERR_MSG in \
+                     open('%s.ipynb' % self._test_name).read() else '0'
+
+    os.chdir(self.TEST_DIR)
+    if self._test_name == 'dsl_static_type_checking':
+      subprocess.call([
+          sys.executable,
+          'check_notebook_results.py',
+          '--testname',
+          self._test_name,
+          '--result',
+          self._sample_test_result,
+          '--exit-code',
+          exit_code
+      ])
+    else:
+      subprocess.call([
+          sys.executable,
+          'check_notebook_results.py',
+          '--testname',
+          self._test_name,
+          '--result',
+          self._sample_test_result,
+          '--namespace',
+          self._namespace,
+          '--exit-code',
+          exit_code
+      ])
+
+    print('Copy the test results to GCS %s/' % self._results_gcs_dir)
+    working_bucket = PROJECT_NAME
+
+    utils.upload_blob(
+        working_bucket,
+        self._sample_test_result,
+        self._results_gcs_dir + '/' + self._sample_test_result
```

Comment:
AFAIK, the results_gcs_dir already contains the whole path including the gs:// and the bucket name. Could we extract the bucket name from the result_gcs_dir?

Reply:
That works. 
My plan is make this extraction logic outside upload_blob() method. Otherwise the upload_blob will keep an assumption that the source file lives under the same bucket as the destination, which is a bit of weird.

Reply:
Agreed that upload_blob should be kept clean and unaware. 
My point was the results_gcs_dir already contains the bucket name, should we extract that as the bucket name(first arg to the upload_blob). It will be confusing later on when the result_gcs_dir contains a non-project-name as the bucket name and the results are in fact stored somewhere else.

Reply:
Done. Extracted the bucket name (project name) from the results_gcs_dir

Reply:
Did you push the commit? I have not seen the changes, yet. 
Thanks